skypy/utils/tests/test_random.py

```diff
@@ -1,52 +1,58 @@
 import numpy as np
-import scipy.stats
-import scipy.integrate
-from skypy.utils import random, special
 
+from scipy.stats import kstest
 
-def test_schechter_cdf():
-    # Test the _schechter_cdf function against precomputed values
-    x = np.logspace(np.log10(0.1), np.log10(10), 10)
-    cdf = random._schechter_cdf(x, min(x), max(x), -1.3)
-    sample = np.array([0., 0.31938431, 0.57066007, 0.75733831, 0.88345969,
-                       0.95631112, 0.98886695, 0.99846589, 0.99992254, 1.])
 
-    np.testing.assert_allclose(cdf, sample)
+def test_schechter():
 
+    from skypy.utils.random import schechter
 
-def test_schechter():
+    def schechter_cdf_gen(alpha, x_min, x_max, resolution=100000):
+        lnx = np.linspace(np.log(x_min), np.log(x_max), resolution)
+
+        cdf = np.exp((alpha + 1)*lnx - np.exp(lnx))
+        np.cumsum((cdf[:-1] + cdf[1:])/2*np.diff(lnx), out=cdf[1:])
+        cdf[0] = 0
+        cdf /= cdf[-1]
+
+        return lambda x: np.interp(np.log(x), lnx, cdf)
```

Comment:
```suggestion
    def schechter_cdf_gen(alpha, x_min, x_max):
        a = special.gammaincc(alpha + 1, x_min)
        b = special.gammaincc(alpha + 1, x)
        c = special.gammaincc(alpha + 1, x_min)
        d = special.gammaincc(alpha + 1, x_max)
        return (a - b) / (c - d)
```
I'd prefer if we stick to testing our "approximate" cdf against the analytic case using incomplete gamma functions.

Reply:
Could you edit the suggestion so that it returns a lambda with a, c, d fixed and computing b for argument x?

Reply:
OK i think I've done that correctly but take a look and check